slack_gpt_bot.py

```diff
@@ -63,8 +65,41 @@ def logging_wrapper(message, severity=logging.INFO, **kwargs):
     func = log.get(severity, json_std_logger.info)
     func(message)
 
+'''
+gpt-3.5-turbo-16k offers 4 times the context length of gpt-3.5-turbo at twice the price: 
+$0.003 per 1K input tokens and $0.004 per 1K output tokens. 
+
+So if the conversation exceeds the cutoff, then switch to using the extended model. Otherwise, use 
+the standard model, as that seems fine for most conversations at this point.
```

Comment:
How does it work if it switches model mid conversation? Does it lose all the context or is that moved over as well?

Reply:
the bot works by using the Slack conversation thread. 
https://github.com/SolarCS/slack-gpt-bot/blob/621cf233e026ffba28c7c3897ec88a0020133241/slack_gpt_bot.py#L137

So it processes the entire conversation in that thread as one message, and sends that to open AI. That's how the bot preserves the context. The difference between the 2 models should only be the token count and cost. Most conversations are brief, so this is a way to extend the conversation without overpaying for the short ones. Or at least that is the idea